### PR TITLE
feat(beacon-node): network worker new space adjustment 

### PR DESCRIPTION
**Pull request by matthewkeil**
**Motivation**

While going through the suggestions that Ben gave us I found that there was a very large amount of new space scavenge collection.  I updated the dashboards and realized it was over 200mb and 3% of the worker thread time spent on scavenge.  There was a lot of page-faults as well which lends itself to a bunch of data being moved during GC.  There was also not very much mark-and-sweep GC which means the objects were most likely getting moved only once before getting un-ref'd. This realization that most of the objects are very short lived, but likely high in volume to trigger the scavenge.  So in essence causing them to get moved back and forth between the `To` and `From` spaces a single time.

I played around with the values of new space to get around this.  I bumped the new space way up above the GC volume.  Now the objects do not live long enough to need to be moved in the new space.  The HandleScope and new space management system are quite efficient so there is little to no performance degradation from leaving the objects in new space once they are no longer ref'd.  They just either sit at the back of the line or get written over when the HandleScope pointers shift.

**Description**

- Updated dashboard panels for memory.
- Added panels for each thread (heap and GC stats) in a row
- Updated network worker new space to a sufficiently large number (over the existing `unstable` GC volume)

[X] - Figure out method to determine how large is too large
[X] - Verify that new space is not too large

Helps with #5604 

** Metrics **

Need to let the node stabilize for a few more hours so traffic is back up to levels on `unstable` for even comparison but things are looking pretty good so far.

Screenshot 2023-08-01 at 3 55 53 AM
Screenshot 2023-08-01 at 3 56 19 AM
Screenshot 2023-08-01 at 3 56 26 AM
Screenshot 2023-08-01 at 3 56 39 AM

Screenshot 2023-08-01 at 4 00 52 AM
Screenshot 2023-08-01 at 4 00 27 AM
Screenshot 2023-08-01 at 3 57 08 AM

Screenshot 2023-08-01 at 4 02 04 AM
Screenshot 2023-08-01 at 4 02 30 AM
Screenshot 2023-08-01 at 4 01 44 AM
Screenshot 2023-08-01 at 4 04 39 AM
Screenshot 2023-08-01 at 4 04 06 AM
Screenshot 2023-08-01 at 4 03 46 AM
Screenshot 2023-08-01 at 4 02 45 AM

### CODE REVIEW

File: packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts [hunk 1]
```diff
@@ -107,7 +107,10 @@ export class WorkerNetworkCore implements INetworkCore {
       loggerOpts: modules.logger.toOpts(),
     };
 
-    const worker = new Worker("./networkCoreWorker.js", {workerData} as ConstructorParameters<typeof Worker>[1]);
+    const worker = new Worker("./networkCoreWorker.js", {
+      workerData,
+      resourceLimits: {maxYoungGenerationSizeMb: 512},
```
Comment (dapplion) [inline packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts]: What's the default value?

**Comment by dapplion** [comment]
Metrics looks promising! This PR does not add metrics, but changes a fundamental parameter, can you split:
- PR A: Add dashboards
- PR B: Change constant value

File: packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts (hunk 1 above)
Comment (matthewkeil) [reply packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts]: It's a [calculated value](https://github.com/nodejs/node/blob/4166d40d0873b6d8a0c7291872c8d20dc680b1d7/deps/v8/src/heap/heap.cc#L266) and ends up roughly 16mb with a `--max-old-space-size` of 4GB.

**Comment by matthewkeil** [comment]
This was updated to just have the single line change.  I rebased this onto unstable from a discussion with @wemeetagain but there is a memory leak on unstable so its hard to see how this is actually performing at 128mb.  I will move to ready for review once we get the leak sorted and can see how this performs with @tuyennhv update with limited subscriptions.

Dashboard changes were moved to PR #5858

**Comment by matthewkeil** [comment]
Value was set from research on #2115.  You can find the resolution to that issue [here](https://github.com/ChainSafe/lodestar/issues/2115#issuecomment-1687860046)


There is a 3x1 ratio of semi space to young generation space in the code but in practice it seems like setting the semi space to the average value of scavenge collection seemed optimal.  In previous empirical testing it seemed to also be the optimal value for the worker young generation setting despite the ratio.  I am not certain of the reason for this but its worth further investigation.

File: packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts [hunk 2]
```diff
@@ -107,7 +107,10 @@ export class WorkerNetworkCore implements INetworkCore {
       loggerOpts: modules.logger.toOpts(),
     };
 
-    const worker = new Worker("./networkCoreWorker.js", {workerData} as ConstructorParameters<typeof Worker>[1]);
+    const worker = new Worker("./networkCoreWorker.js", {
+      workerData,
+      resourceLimits: {maxYoungGenerationSizeMb: 152},
```
Comment (wemeetagain) [inline packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts]: Can you add a comment here linking / describing why 152?

Comment (wemeetagain) [reply packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts]: might be worth exposing as a hidden cli flag so we don't need to PR to test other values

Comment (matthewkeil) [reply packages/beacon-node/src/network/core/networkCoreWorkerHandler.ts]: Updated.

**Review by wemeetagain — DISMISSED** [review]

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.11.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.11.0) :tada:

**Pull request merged**